packages/moonstone/Dropdown/Dropdown.js

```diff
@@ -140,6 +151,54 @@ const DropdownList = Skinnable(
 	})
 );
 
+class DropdownList extends React.Component {
+	static displayName = 'DropdownList'
+
+	static propTypes = {
+		/*
+		 * Index of the selected item.
+		 *
+		 * @type {Number}
+		 */
+		selected: PropTypes.number
+	}
+
+	componentDidMount () {
+		// eslint-disable-next-line react/no-find-dom-node
+		this.node = ReactDOM.findDOMNode(this);
+		Spotlight.set(this.node.dataset.spotlightId, {leaveFor: {up: '', down: ''}});
+	}
+
+	componentDidUpdate () {
+		if (isSelectedValid(this.props)) {
```

Comment:
I'm assuming this should only on open so we should add a guard to that end. I would suggest refactoring into a new method too. We seem to muddy up lifecycle methods over time by mixing concerns and starting with separate methods will help future us keep things clear.

Reply:
Point taken about refactoring into a new method.

I didn't add the guard for `open`, since the list component is only mounted/updated in the popup when the list is open, so it seemed redundant. Are there cases where it is still needed?

Reply:
It's unlikely but possible to break it. To test it, I added a `setInterval` in a view holding a drop down and used it to toggle the `width` every half second. When the popup was open, focus would get reset and rescrolled each time.